pandas/core/resample.py

```diff
@@ -1043,6 +1043,7 @@ def __init__(self, obj, parent=None, groupby=None, **kwargs):
         # the resampler attributes
         for attr in self._attributes:
             setattr(self, attr, kwargs.get(attr, getattr(parent, attr)))
+        self._selection = kwargs.get("selection")
```

Comment:
can this be done in stead in Resampler? this is quite low-level and _selection is note event defined here

Reply:
_GroupByMixin is only used for (Datetime|Period|Timedelta)IndexResamplerGroupBy, so this change is localized.

It doesn't appear to me this can be moved to Resampler without quite a bit of hacking. The issue is that DatetimeIndexResamplerGroupby inherits both from _GroupByMixin and DatetimeIndexResampler, with _GroupByMixin first in the MRO. So currently, when constructing a new DatetimeIndexResamplerGroupby, the init of DatetimeIndexResampler (and hence, Resampler) never gets called.

Both _GroupByMixin and Resampler inits take a groupby argument, but they do not mean the same thing.  _GroupByMixin stashes groupby in `self._groupby` whereas Resampler in `self.groupby`. Moreover,  _GroupByMixin uses `parent.groupby` for `self.groupby`.

I've added _selection as an instance attribute via type-hints.

Reply:
ok great. yeah this whole structure is not entirely great, but sounds good for now.